File: tools/fix-ide.cs
using System;
using System.IO;
using System.Linq;

class Program
{
    private const string OldLine = @"            ""$output"": ""-MMD -o ${out} ${in}"",";
    private const string NewLine = @"            ""$output"": "" -o ${out} ${in}"",";

    static int Main(string[] args)
    {
        bool automaticmode = false;
        if (args.Length != 2)
        {
            Console.WriteLine("Usage: program.exe <directory_path> <filename>");
            Console.WriteLine("no arguments provided, attempting automatic mode...");
            automaticmode = true;
        }



        if (automaticmode)
        {
            string vscodePath;

            if (getVsCodePath(out vscodePath))
            {
                Console.WriteLine("Found vscode path: " + vscodePath);
                try
                {
                    int modifiedCount = ProcessFiles(vscodePath, "sdcc.model.json");
                    Console.WriteLine($"\nCompleted! Modified {modifiedCount} files.");
                    return 0;
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"An error occurred: {ex.Message}");
                    return 1;
                }
                Console.WriteLine($"\nCompleted! Modified {modifiedCount} files.");
            }

            if (getVsCodeInsidersPath(out vscodePath))
            {
                Console.WriteLine("Found vscode path: " + vscodePath);
                try
                {
                    int modifiedCount = ProcessFiles(vscodePath, "sdcc.model.json");
                    Console.WriteLine($"\nCompleted! Modified {modifiedCount} files.");
                    return 0;
                }
                catch (Exception ex)
                {
                    Console.WriteLine($"An error occurred: {ex.Message}");
                    return 1;
                }
                Console.WriteLine($"\nCompleted! Modified {modifiedCount} files.");
            }
        }
        else
        {
            string directoryPath = args[0];
            string filename = args[1];

            if (!Directory.Exists(directoryPath))
            {
                Console.WriteLine($"Error: Directory '{directoryPath}' does not exist");
                return 1;
            }

            try
            {
                int modifiedCount = ProcessFiles(directoryPath, filename);
                Console.WriteLine($"\nCompleted! Modified {modifiedCount} files.");
                return 0;
            }
            catch (Exception ex)
            {
                Console.WriteLine($"An error occurred: {ex.Message}");
                return 1;
            }
        }

        return 0;
    }

    private static int ProcessFiles(string directoryPath, string filename)
    {
        int modifiedCount = 0;

        // Find all matching files recursively
        var files = Directory.GetFiles(directoryPath, filename, SearchOption.AllDirectories);

        foreach (var file in files)
        {
            Console.WriteLine($"Processing: {file}");
            if (ProcessFile(file))
            {
                modifiedCount++;
                Console.WriteLine($"Modified: {file}");
            }
        }

        return modifiedCount;
    }

    private static bool getVsCodePath(out string path)
    {
        //get the vscode paths .vscode or .vscode-insiders
        string vscodePath = Path.Combine(Environment.GetEnvironmentVariable("userprofile"), ".vscode", "extensions");

        if (Directory.Exists(vscodePath))
        {
            path = vscodePath;
            return true;
        }
        else
        {
            path = "";
            return false;
        }
    }

    private static bool getVsCodeInsidersPath(out string path)
    {

        //get the vscode paths .vscode or .vscode-insiders
        string vscodePath = Path.Combine(Environment.GetEnvironmentVariable("userprofile"), ".vscode-insiders", "extensions");
        if (Directory.Exists(vscodePath))
        {
            path = vscodePath;
            return true;
        }
        else
        {
            path = "";
            return false;
        }
    }


    private static bool ProcessFile(string filePath)
    {
        string tempPath = filePath + ".tmp";
        bool modified = false;

        try
        {
            // Read all lines and modify as needed
            var lines = File.ReadAllLines(filePath);
            for (int i = 0; i < lines.Length; i++)
            {
                if (lines[i] == OldLine)
                {
                    lines[i] = NewLine;
                    modified = true;
                }
            }

            // Only write if modifications were made
            if (modified)
            {
                File.WriteAllLines(tempPath, lines);
                File.Delete(filePath);
                File.Move(tempPath, filePath);
            }

            return modified;
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Error processing {filePath}: {ex.Message}");
            if (File.Exists(tempPath))
            {
                File.Delete(tempPath);
            }
            return false;
        }
    }
}

File: tools/fix-ide.py
import os
import sys
import fileinput
from pathlib import Path

def replace_in_file(file_path):
    """Replace the specified line in a file and return True if changes were made."""
    old_line = '            "$output": "-MMD -o ${out} ${in}",'
    new_line = '            "$output": " -o ${out} ${in}",'
    
    changed = False
    temp_file = str(file_path) + '.tmp'
    
    try:
        with open(file_path, 'r') as file, open(temp_file, 'w') as temp:
            for line in file:
                if line.rstrip() == old_line:
                    temp.write(new_line + '\n')
                    changed = True
                else:
                    temp.write(line)
                    
        if changed:
            os.replace(temp_file, file_path)
        else:
            os.remove(temp_file)
            
        return changed
    
    except Exception as e:
        print(f"Error processing {file_path}: {str(e)}")
        if os.path.exists(temp_file):
            os.remove(temp_file)
        return False

def main():
    if len(sys.argv) != 3:
        print(f"Usage: {sys.argv[0]} <directory_path> <filename>")
        sys.exit(1)
        
    directory = Path(sys.argv[1])
    filename = sys.argv[2]
    
    if not directory.is_dir():
        print(f"Error: Directory '{directory}' does not exist")
        sys.exit(1)
    
    modified_count = 0
    
    # Find all matching files and process them
    for file_path in directory.rglob(filename):
        print(f"Processing: {file_path}")
        if replace_in_file(file_path):
            modified_count += 1
            print(f"Modified: {file_path}")
    
    print(f"\nCompleted! Modified {modified_count} files.")

if __name__ == "__main__":
    main()